### ISSUE TITLE
Best way to handle error messages?

### ISSUE THREAD

**Issue by markusrubey** [where2help/where2help#36]
##### Error responses vary by endpoint and deliver a mixed bag of german and english

**auth/** is quite nice because of the detailed (and from the user understandable) infos inside the ```"full_messages"``` array.
```json
  "errors": {
    "password": ["ist zu kurz (weniger als 8 Zeichen)"],
    "email": ["address is already in use"],
    "full_messages": ["Passwort ist zu kurz (weniger als 8 Zeichen)", "E-Mail address is already in use"]
  }
```
**auth/sign_in** on the other hand just returns "errors"
```json
  "errors": ["translation missing: de.devise_token_auth.sessions.bad_credentials"]
```
<br>

So an option would be to use the error messages directly (But what about translation?):
```json
  "errors": ["Passwort ist zu kurz (weniger als 8 Zeichen)", "E-Mail ist bereits vergeben"]
```
Or I assign my own set of strings to the keywords. (Would need a list of all the keywords though)
```json
  "errors": {
    "password": ["too_short", "no_match", ...],
    "email": ["taken", "malformed", ...]
  }
```

**Comment by haslinger** [comment]
Rails is just missing the translations, I'll add them.

**Comment by markusrubey** [comment]
Thanks! Tomorrow eh? ;-) And how would I switch between languages? Just adding a local header or something?

**Comment by haslinger** [comment]
I translated the gem to German and requested a pull there, but to be able to continue, I just included the translations here in our app. Pushed to staging as well.
 
If you want to know what to expect when: https://github.com/where2help/where2help/blob/master/config/locales/devise/auth_token_de.yml

As we don't have any information in the database that is multilingual, switching the language for an API request does not make any sense currently. You would get the same payload anyways. 
The translations above are only for Interface elements, not for actual data.

If we had it, we would provide you data in attribute_name_de and attribute_name_en and whatever instead of attribute_name and you just pick, what you like.
